sstables/compress.cc

```diff
@@ -157,7 +157,9 @@ void compression::segmented_offsets::state::update_position_trackers(std::size_t
 }
 
 void compression::segmented_offsets::init(uint32_t chunk_size) {
-    SCYLLA_ASSERT(chunk_size != 0);
+    if (chunck_size == 0) {
+        throw sstables::malformed_sstable_exception("Segmented offsets chunk size is zero.");
```

Comment:
Need to add sstable filename here.

Reply:
Output after corruption:
`Startup failed: std::runtime_error (Exception while populating keyspace 'system' with column family 'cdc_generations_v3' from datadir 'tmp/data/system/cdc_generations_v3-c697df3f55393ef5b003843be7ba1422': sstables::malformed_sstable_exception (Buffer improperly sized to hold requested data. Got: 151. Expected: 12336 in sstable tmp/data/system/cdc_generations_v3-c697df3f55393ef5b003843be7ba1422/me-3gk0_0xu7_34ycw248ycqx6sa8nv-big-CompressionInfo.db))
`

Reply:
Turns out we already check this in the layer above, in sstables::parse().